ios/brave-ios/Sources/Brave/Frontend/Browser/New Tab Page/Backgrounds/NewTabPageVideoButtonsView.swift

```diff
@@ -0,0 +1,199 @@
+      target: self,
+      action: #selector(self.videoTapped(sender:))
+    )
+    tapGesture.numberOfTapsRequired = 1
+    addGestureRecognizer(tapGesture)
+  }
+
+  private func tappedVideoCancelButton() {
+    tappedCancelButton?()
+  }
+
+  @objc private func videoTapped(sender: UITapGestureRecognizer) {
+    let location = sender.location(in: self)
+    if let view = super.hitTest(location, with: nil), view == cancelButton {
+      tappedVideoCancelButton()
+      return
+    }
+
+    guard let playStarted = tappedBackground?() else {
+      return
+    }
+
+    if playStarted {
+      playButtonImage.isHidden = false
+      pauseButtonImage.isHidden = true
+      showAndFadeOutImage(imageView: playButtonImage)
+    } else {
+      pauseButtonImage.isHidden = false
+      playButtonImage.isHidden = true
+      showAndFadeOutImage(imageView: pauseButtonImage)
+    }
+  }
+
+  private func showAndFadeOutImage(imageView: UIView) {
+    imageView.alpha = 0
+    UIView.animate(
+      withDuration: 0.1,
+      animations: {
+        imageView.alpha = 1
+      },
+      completion: { _ in
+        UIView.animate(
+          withDuration: 0.3,
+          delay: 0.5,
+          animations: {
+            imageView.alpha = 0
+          }
+        )
+      }
+    )
+  }
+
+  @available(*, unavailable)
+  required init(coder: NSCoder) {
+    fatalError()
+  }
+
+  override func layoutSubviews() {
+    super.layoutSubviews()
+
+    cancelButton.snp.remakeConstraints {
+      $0.top.equalTo(self.safeAreaLayoutGuide.snp.top).offset(20)
+      $0.right.equalTo(self.safeAreaLayoutGuide.snp.right).offset(-20)
+    }
+
+    playButtonImage.snp.remakeConstraints {
+      $0.centerX.equalToSuperview()
+      $0.centerY.equalToSuperview().offset(20)
+    }
+
+    pauseButtonImage.snp.remakeConstraints {
+      $0.centerX.equalToSuperview()
+      $0.centerY.equalToSuperview().offset(20)
+    }
+  }
+}
+
+extension NewTabPageVideoButtonsView {
+  private class CancelButton: SpringButton {
+    let imageView = UIImageView(
+      image: UIImage(named: "ntt_cancel_button", in: .module, compatibleWith: nil)!
+    )
+
+    private let backgroundView = UIVisualEffectView(
+      effect: UIBlurEffect(style: .systemThinMaterialDark)
+    ).then {
+      $0.clipsToBounds = true
+      $0.isUserInteractionEnabled = false
+    }
+
+    override init(frame: CGRect) {
+      super.init(frame: frame)
+
+      clipsToBounds = true
+
+      addSubview(backgroundView)
+
+      backgroundView.contentView.addSubview(imageView)
+
+      backgroundView.snp.makeConstraints {
+        $0.edges.equalToSuperview()
+      }
+
+      imageView.snp.makeConstraints {
+        $0.edges.equalToSuperview().inset(UIEdgeInsets(equalInset: 4))
+      }
+    }
+
+    override func layoutSubviews() {
+      super.layoutSubviews()
+
+      layer.cornerRadius = bounds.height / 2.0
+    }
+  }
+
+  private class BluredImageView: UIView {
```

Comment:
Can just use the generic button mentioned above and set `isEnabled` on it to `false`